### ISSUE TITLE
[Doc]: Documenting rcParam default values

### ISSUE THREAD

**Issue by oscargus** [matplotlib/matplotlib#25644]
### Documentation Link

https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.html#matplotlib.axes.Axes

### Problem

There is a nice, automatically, generated kwargs section. However, it is not clear which values have defaults that are controlled by rcParams, nor what they are.

### Suggested improvement

If I understand correctly, this is generated from the documentation of the corresponding `set_*` methods. An option may be to add a decorator, e.g. `@rcparam('axes.facecolor')` which adds a third column(?) (Default?) with the contents of `:rc:...`.

(Naturally, not just for `Axes`.)

**Comment by timhoffm** [comment]
Related approach: #22699. Or likely better the general artist_property decorator approach as suggested in #22749.

Feel free to take over as I don’t have the time to push that.

**Comment by oscargus** [comment]
Thanks for remining me about those! That is definitely also a way forward!

**Comment by Higgs32584** [comment]
If I am mistaken, please let me know, but would a potential solution to this problem by changing the default comments in  
C:\Users\mjhig\.vscode\bob\matplotlib\lib\matplotlib\axes\_base.py? Or would it involve some kind of appending an index.rst to replace the auto-generation? Thank you!

**Comment by oscargus** [comment]
The problem is that for the different `set_*`-methods, it is not always possible to pass `None` to restore the default rcParam behavior. Also, for some properties, the type is "unknown" (for unknown reasons, despite being specified in the corresponding `set_*`-method.

If I understand your question correctly.

I think my suggestion is to extend the auto-generation to include a third column, but when being reminded about the other options, I believe that one of the previous approaches is better/can be combined with this. (As they would also solve the "unknown" issue and result in less parsing so faster doc-build times.)

**Comment by timhoffm** [comment]
I agree that the artist property tables are not the most informative. "Description" is mostly the accepted type.

However the underlying problem is that artist properties are only implicitly defined: They are the things you can `set_*`. Any information is inferred dynamically. In particular the "default value" is just what the object is created with and that can be specified be anywhere from `__init__`, some helper functions, `set_*` itself in the class or base classes. There is no way we can infer this automatically and not even a reasonable place we could attach this info to if we'd want to define this explicitly.

Due to the current state, I'd say it's rightfully closed as not planned. If we wanted to move kwargs documentation, I think it would necessarily go though formalizing the concept of Artist properties; e.g. something like #22749.
